### PR TITLE
Implemented proto translation for IP announcement messages

### PR DESCRIPTION
**Pull request by pompon0**

### CODE REVIEW

File: chain/network/src/network_protocol/proto_conv/handshake.rs [hunk 1]
```diff
@@ -0,0 +1,185 @@
+/// Conversion functions for Hanshake messages.
```
Comment (jakmeier) [inline chain/network/src/network_protocol/proto_conv/handshake.rs:1]:
```suggestion
/// Conversion functions for handshake messages.
```

File: chain/network/src/network_protocol/proto_conv/util.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#[derive(thiserror::Error, Debug)]
+#[error("[{idx}]: {source}")]
+pub struct ParseVecError<E> {
+    idx: usize,
+    #[source]
+    source: E,
+}
+
+pub fn try_from_vec<'a, X, Y: TryFrom<&'a X>>(
+    xs: &'a Vec<X>,
```
Comment (jakmeier) [inline chain/network/src/network_protocol/proto_conv/util.rs:13]: Unless I am missing something, this looks like it want to be:
```suggestion
    xs: &'a [X],
```
(There are very few reasons to ever use `&Vec<_>` in a parameter, slices are just more versatile.)

**Review by jakmeier — COMMENTED** [review]
Not sure if the code review request was intentional here. I saw too late that this is not a PR against the master branch, so anyway, here are a few review comments. ;)

File: chain/network/src/network_protocol/proto_conv/account_key.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                .map_err(Self::Error::Timestamp)?,
+        })
+    }
+}
+
+//////////////////////////////////////////
+
+// TODO: I took this number out of thin air,
+// determine a reasonable limit later.
+const VALIDATOR_PAYLOAD_MAX_BYTES: usize = 10000;
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/network_protocol/proto_conv/account_key.rs:65]: Nope, I have no idea what is the right limit at the moment, or either it should this constraint be validated here, or up the stack (see another TODO below). The thing is that this message type is not in use yet. I'll calibrate it once I have everything plugged end-to-end. I've expanded the TODO description a bit and added a relevant jira issue.

File: chain/network/src/network_protocol/proto_conv/util.rs (hunk 2 above)
Comment (pompon0) [reply chain/network/src/network_protocol/proto_conv/util.rs:13]: done

File: chain/network/src/network_protocol/borsh_conv.rs [hunk 4]
```diff
@@ -92,7 +106,9 @@ impl TryFrom<&net::PeerMessage> for mem::PeerMessage {
                 mem::PeerMessage::HandshakeFailure(pi, (&hfr).into())
             }
             net::PeerMessage::LastEdge(e) => mem::PeerMessage::LastEdge(e),
-            net::PeerMessage::SyncRoutingTable(rtu) => mem::PeerMessage::SyncRoutingTable(rtu),
+            net::PeerMessage::SyncRoutingTable(rtu) => {
+                mem::PeerMessage::SyncRoutingTable((&rtu).into())
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/network_protocol/borsh_conv.rs:110]: done

File: chain/network/src/network_protocol/proto_conv/handshake.rs (hunk 1 above)
Comment (jakmeier) [reply chain/network/src/network_protocol/proto_conv/handshake.rs:1]: I'd still like to see the "d" though :)

(Personally, I either write things in plain English or in `formatted_code` but I honestly don't care. I only nitpicked here as I saw two things in one word.)

File: chain/network/src/network_protocol/proto_conv/account_key.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (jakmeier) [reply chain/network/src/network_protocol/proto_conv/account_key.rs]: A linked Jira issue already makes me much happier

**Review by jakmeier — APPROVED** [review]
LGTM

File: chain/network/src/network_protocol/proto_conv/handshake.rs (hunk 1 above)
Comment (pompon0) [reply chain/network/src/network_protocol/proto_conv/handshake.rs:1]: > I'd still like to see the "d" though :)

you got me XD
